include/evmc/helpers.hpp

```diff
@@ -53,17 +53,35 @@ inline bool is_zero(const evmc_bytes32& x) noexcept
     return x == evmc_bytes32{};
 }
 
-/// FNV1a hash function with 64-bit result.
-inline uint64_t fnv1a_64(const uint8_t* ptr, size_t len)
+template <size_t size>
+struct fnv1_params
 {
-    constexpr uint64_t prime = 1099511628211ULL;
-    constexpr uint64_t offset_basis = 14695981039346656037ULL;
+};
+
+template <>
+struct fnv1_params<4>
+{
+    static constexpr auto prime = 0x1000193;
+    static constexpr auto offset_basis = 0x811c9dc5;
+};
+
+template <>
+struct fnv1_params<8>
+{
+    static constexpr auto prime = 0x100000001b3;
+    static constexpr auto offset_basis = 0xcbf29ce484222325;
```

Comment:
Are these the same as the decimals in the previous version?

Reply:
Yes. The `0x...` makes them `unsigned`.

Reply:
The decimal version had a `ULL` postfix which supposedly would make it unsigned.

Reply:
Yes, but then you have to decide what type it has to be unless `u` postfix will make it `unsigned long` if big enough.